### PR TITLE
Mod/10468 class to fn recipient filter

### PR DESCRIPTION
**Pull request by brianpetway**
**High level description:**

Change these from class to functional components

**Technical details:**

I did that. I deleted RecipientToggle bc it wasn't used anywhere.

**JIRA Ticket:**
[DEV-10468](https://federal-spending-transparency.atlassian.net/browse/DEV-10468)

**Mockup:**
n/a

The following are ALL required for the PR to be merged:

Author:
- [x ] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [ ] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

**Review by anjenkin — CHANGES_REQUESTED** [review]
see comments inline

File: src/js/components/search/filters/recipient/RecipientSearch.jsx [hunk 1]
```diff
@@ -16,39 +16,50 @@ const propTypes = {
...
-                    <SubmitHint
-                        ref={(component) => {
-                            this.hint = component;
-                        }} />
-                </div>
+    return (
+        <div className="recipient-filter">
+            <div className="filter-item-wrap">
+                <RecipientNameDUNSContainer
+                    selectedRecipients={selectedRecipients}
```
[the remark this answers is not in the document]
Comment (brianpetway) [reply src/js/components/search/filters/recipient/RecipientSearch.jsx:52]: Yes, this was intentional. Some props were not actually used so I removed them.

File: src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx [hunk 2]
```diff
@@ -78,81 +78,87 @@ const propTypes = {
...
                 </div>
+                <RecipientTypeList
+                    expanded={expanded?.includes(category.id)}
+                    selectedTypes={selectedTypes}
+                    category={category}
+                    toggleCheckboxType={toggleCheckboxType}
+                    recipientTypes={recipientTypes} />
+            </div>));
+
+    const usePrevious = (value) => {
```
[the remark this answers is not in the document]
Comment (brianpetway) [reply src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx:129]: Same answer as above

File: src/js/components/search/filters/recipient/RecipientSearch.jsx [hunk 3]
```diff
@@ -16,39 +16,50 @@ const propTypes = {
...
+
+    if (selectedRecipients.size > 0) {
+        localSelectedRecipients = (<SelectedRecipients
+            selectedRecipients={selectedRecipients}
+            toggleRecipient={toggleRecipient} />);
     }
 
-    render() {
-        let selectedRecipients = null;
+    const usePrevious = (value) => {
```
[the remark this answers is not in the document]
Comment (brianpetway) [reply src/js/components/search/filters/recipient/RecipientSearch.jsx:30]: I will use that helper instead.

**Review by anjenkin — APPROVED** [review]

**Pull request merged**